mate-settings-daemon/main.c

```diff
@@ -312,79 +179,63 @@ watch_for_term_signal (MateSettingsManager *manager)
+                                                  G_DBUS_CALL_FLAGS_NONE,
+                                                  -1,
+                                                  NULL,
+                                                  &error);
+                if (variant == NULL) {
+                        g_warning ("Could not ask session manager to log out: %s", error->message);
                         g_error_free (error);
-                }
-                else {
-                        /* get org.gnome.SessionManager.ClientPrivate interface */
-                        private_proxy = dbus_g_proxy_new_for_name_owner (bus, MATE_SESSION_DBUS_NAME,
-                                                                         client_id, MATE_SESSION_PRIVATE_DBUS_INTERFACE,
-                                                                         &error);
+                } else {
+                        g_variant_get (variant, "(o)", &client_id);
+                        g_variant_unref (variant);
+
+                        private_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
+                                                                       G_DBUS_PROXY_FLAGS_NONE,
+                                                                       NULL,
+                                                                       MATE_SESSION_DBUS_NAME,
+                                                                       client_id,
+                                                                       MATE_SESSION_PRIVATE_DBUS_INTERFACE,
+                                                                       NULL,
+                                                                       &error);
                         if (private_proxy == NULL) {
                                 g_warning ("DBUS error: %s", error->message);
                                 g_error_free (error);
                         }
-                        else {
-
-                            /* get QueryEndSession */
-                            dbus_g_proxy_add_signal (private_proxy, "QueryEndSession", G_TYPE_UINT, G_TYPE_INVALID);
-                            dbus_g_proxy_connect_signal (private_proxy, "QueryEndSession",
-                                                         G_CALLBACK (on_session_query_end),
-                                                         manager, NULL);
-
-                            /* get EndSession */
-                            dbus_g_proxy_add_signal (private_proxy, "EndSession", G_TYPE_UINT, G_TYPE_INVALID);
-                            dbus_g_proxy_connect_signal (private_proxy, "EndSession",
-                                                         G_CALLBACK (on_session_end), manager, NULL);
-
-                        }
 
+                        g_signal_connect (G_OBJECT (private_proxy), "g-signal", G_CALLBACK (on_private_signal), manager);
```

Comment:
Readability: the line length is too wide making it difficult to read.

Reply:
It’s an improvement, but to make it even easier to read, use an argument on each line.